internal/process/defunct_processes.go

```diff
@@ -0,0 +1,25 @@
+package process
+
+import (
+	"github.com/shirou/gopsutil/process"
+)
+
+// DefunctProcesses returns all the defunct processes in the node
+func DefunctProcesses() ([]*process.Process, error) {
+	processes, err := process.Processes()
+	if err != nil {
+		return nil, err
+	}
+
+	defunctProcesses := []*process.Process{}
+
+	for _, p := range processes {
+		var status string
+		status, err = p.Status()
+
+		if err == nil && status == "Z" {
+			defunctProcesses = append(defunctProcesses, p)
+		}
+	}
+	return defunctProcesses, nil
+}
```

Comment:
Thank you @kolyshkin and @saschagrunert for your advice. I wiped out everything I had done in `internal/process/defunct_processes.go` earlier and used my own implementation where I am only returning the PIDs of all the zombie processes. Would greatly appreciate your input before I make another commit :)

```suggestion
package process

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

// DefunctProcesses returns the PIDs of all the zombie processes in the node
func DefunctProcesses() ([]int, error) {
	directories, err := os.Open("/proc")
	if err != nil {
		return nil, err
	}
	defer directories.Close()

	names, err := directories.Readdirnames(-1)
	if err != nil {
		return nil, err
	}

	var pids []int
	for _, name := range names {
		// processes have numeric names. If the name does not start with a number, skip.
		if name[0] < '0' || name[0] > '9' {
			continue
		}

		pid, err := strconv.ParseInt(name, 10, 0)
		if err != nil {
			continue
		}

		stat, err := status(int(pid))
		if err != nil {
			continue
		}
		if stat == "Z" {
			pids = append(pids, int(pid))
		}
	}
	return pids, nil
}

// status returns the status of a process
func status(pid int) (string, error) {
	bytes, err := os.ReadFile("/proc" + strconv.Itoa(pid) + "stat")
	if err != nil {
		return "", err
	}
	data := string(bytes)

	i := strings.LastIndexByte(data, ')')
	if i <= 2 || i >= len(data)-1 {
		return "", fmt.Errorf("invalid stat data (no comm): %q", data)
	}
	return string(data[i+2]), nil
}
```

Reply:
@raisaat please commit your suggestion, this way we can better comment line-by-line. :)

Reply:
@saschagrunert made a new commit :)